### ISSUE TITLE
JSON file (from JSONFormatter) does not contain current feature results/data if run is aborted.

### ISSUE THREAD

**Issue by wombat70** [behave/behave#842]
### tl;dr - JSON file (from JSONFormatter) does not contain current feature results/data if run is aborted.

I recently had a need to abort test runs in case of certain step/scenario failures - i.e. not necessarily on first failure (which I could do with `--stop`). I simply used the `KeyboardInterrupt()` exception mechanism in my `after_scenario` hook, for example:

```
def after_scenario(context, scenario):
    if (scenario.status == behave_status.failed) and ('abort_on_fail' in context.tags):
        raise KeyboardInterrupt()
```

This worked fine, except that I also like to use the JSONFormatter (`--formatter json.pretty`) and I noticed that if a run is interrupted/aborted like this, then **I get a JSON file which excludes the current (failed/aborted) feature.** _If I run with multiple features, I only get JSON results for the preceding (passing) features._

### Steps to reproduce.

**features/abort.feature** 
```
Feature: Test behave abort behaviour.

  In order to be able to debug an application
  As a test engineer
  I want to be able to abort a test run and get a complete JSON result file.

  @abort_on_fail
  Example: This failing scenario is tagged to abort on failure.
    Given this step passes
    When this step fails
    Then this step passes

```

**features/steps/abort_steps.py**
```
#!/bin/python
from behave import given, when, then, step

@given(u'this step passes')
@then(u'this step passes')
def step_impl(context):
    pass

@when(u'this step fails')
def step_impl(context):
    assert False, 'Step failed!'
```

**features/environment.py**
```
#!/bin/python
from behave.model_core import Status as behave_status
def after_scenario(context, scenario):
  if (scenario.status == behave_status.failed) and ('abort_on_fail' in context.tags):
    raise KeyboardInterrupt()
```

**When run with the JSONFormatter, I get an empty JSON file:**
```
$ behave features/abort.feature --format json.pretty --outfile result.json

ABORTED: By user.

Failing scenarios:
  features/abort.feature:8  This failing scenario is tagged to abort on failure.

0 features passed, 1 failed, 0 skipped
0 scenarios passed, 1 failed, 0 skipped
1 step passed, 1 failed, 1 skipped, 0 undefined
Took 0m0.000s
```

**result.json**
```
[

]
```

**Comment by wombat70** [comment]
#843
